### PR TITLE
Rails 5.1.4 upgrade

### PR DESCRIPTION
**Pull request by motizuki**
The Worker is upgraded from 4.1.4 to Rails 5.1.

See release notes.
https://github.com/DigitalNZ/supplejack_worker/releases/tag/untagged-2f65b323c696174eb5a0

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,68 +1,49 @@
...
-
-gem 'supplejack_common', git: 'https://github.com/DigitalNZ/supplejack_common.git'
-# gem 'supplejack_common', path: '/webspace/supplejack/common'
-#gem 'supplejack_common', path: ENV['SUPPLEJACK_COMMON']
+gem 'rails', '~> 5.1.4'
+gem 'puma', '~> 3.7'
 
-gem 'rails', '4.1.4'
-gem 'protected_attributes'
+gem 'supplejack_common', path: '../supplejack_common'
```
Comment (Reinkaos) [inline Gemfile]: point to rails 5 branch

File: app/controllers/collection_statistics_controller.rb [hunk 2]
```diff
@@ -7,13 +7,18 @@
...
+      @collection_statistics = CollectionStatistics.where(collection_statistics_params)
     else
       @collection_statistics = CollectionStatistics.all
     end
     render json: @collection_statistics
   end
 
+  private
+
+  def collection_statistics_params
```
Comment (Reinkaos) [inline app/controllers/collection_statistics_controller.rb]: get the right params permitted

File: config/application.yml.example [hunk 3]
```diff
@@ -1,61 +0,0 @@
-# Add account credentials and API keys here.
```
Comment (Reinkaos) [inline config/application.yml.example]: create a new sample yml file with correct data

File: config/environments/production.rb [hunk 4]
```diff
@@ -1,84 +1,76 @@
...
 
-  # Disable delivery errors, bad email addresses will be ignored
+  # Ignore bad email addresses and do not raise email delivery errors.
+  # Set this to true and configure the email server for immediate delivery to raise delivery errors.
   # config.action_mailer.raise_delivery_errors = false
 
-  # Do not eager load code on boot.
-  config.eager_load = true
-
-  ActionMailer::Base.delivery_method = :smtp
```
Comment (Reinkaos) [inline config/environments/production.rb]: add smtp server config - check if it actually needs email

File: config/environments/staging.rb [hunk 5]
```diff
@@ -1,84 +0,0 @@
-# The Supplejack Worker code is Crown copyright (C) 2014, New Zealand Government,
```
Comment (Reinkaos) [inline config/environments/staging.rb]: we need a staging config :)

File: config/initializers/kaminari_config.rb [hunk 6]
```diff
@@ -1,17 +1,12 @@
...
-# and is licensed under the GNU General Public License, version 3. 
-# See https://github.com/DigitalNZ/supplejack_worker for details. 
-# 
-# Supplejack was created by DigitalNZ at the National Library of NZ
-# and the Department of Internal Affairs. http://digitalnz.org/supplejack
-
+# frozen_string_literal: true
 Kaminari.configure do |config|
-  config.default_per_page = 20
+  # config.default_per_page = 25
```
Comment (Reinkaos) [inline config/initializers/kaminari_config.rb]: change back to 20

File: config/initializers/lograge.rb [hunk 7]
```diff
@@ -1,11 +0,0 @@
-HarvesterWorker::Application.configure do
```
Comment (Reinkaos) [inline config/initializers/lograge.rb]: check with Dan if we are using lograge

File: app/controllers/harvest_jobs_controller.rb [hunk 8]
```diff
@@ -11,7 +11,7 @@ class HarvestJobsController < ApplicationController
   before_action :authenticate_user!
 
   def index
-    @harvest_jobs = HarvestJob.search(params)
+    @harvest_jobs = HarvestJob.search(params.permit!)
```
Comment (Reinkaos) [inline app/controllers/harvest_jobs_controller.rb]: unsafe params

File: app/models/abstract_job.rb [hunk 9]
```diff
@@ -45,7 +45,7 @@ class AbstractJob
   scope :disposable, -> { lt(created_at: Time.now - 3.months) }
 
   def self.search(params)
-    search_params = params.try(:dup).try(:symbolize_keys) || {}
+    search_params = params.to_h.try(:dup).try(:symbolize_keys) || {}
```
Comment (Reinkaos) [inline app/models/abstract_job.rb]: maybe dup doesn't need a try

File: Gemfile (hunk 1 above)
Comment (motizuki) [reply Gemfile]: fixed in the further commit

File: config/application.yml.example (hunk 3 above)
Comment (motizuki) [reply config/application.yml.example]: restored

File: config/environments/production.rb (hunk 4 above)
Comment (motizuki) [reply config/environments/production.rb]: smtp added

File: app/models/abstract_job.rb (hunk 9 above)
Comment (motizuki) [reply app/models/abstract_job.rb]: done

File: config/initializers/lograge.rb (hunk 7 above)
Comment (motizuki) [reply config/initializers/lograge.rb]: We are going to use the default logging from Rails 5 now and see how it goes.

**Pull request merged**